userspace/libsinsp/container.cpp

```diff
@@ -489,72 +635,90 @@ bool sinsp_container_engine_docker::resolve(sinsp_container_manager* manager, si
 		return false;
 
 	tinfo->m_container_id = container_info.m_id;
```

Comment:
As we were discussing on slack, it's probably better to set the threadinfo's container id *only* when the full container info is available. This is important for falco for rules like `evt.type=execve and proc.vpid=1 and container.id=host and <other container info checks>`. If the container id were set too early, the rule would start trying to look up container information before it really was available.